subiquity/server/server.py

```diff
@@ -552,8 +553,93 @@ async def apply_autoinstall_config(self, context):
             await controller.apply_autoinstall_config()
             await controller.configured()
 
+    def _strip_controller_keys(
+        self,
+        autoinstall_config: dict[str, Any],
+    ) -> dict[str, Any]:
+        """Removes all unkown keys from autoinstall config."""
+
+        return_dict: dict = copy.deepcopy(autoinstall_config)
+
+        for controller in self.controllers.instances:
+            return_dict.pop(controller.autoinstall_key, None)
+
+        return return_dict
+
+    @with_context(name="top_level_keys")
+    def _enforce_top_level_keys(
+        self,
+        *,
+        autoinstall_config: dict[str, Any],
+        context: Context,
+    ) -> dict[str, Any]:
+        """Enforces usage of known top-level keys.
+
+        In Autoinstall v1, unknown top-level keys are removed from
+        the config and a cleaned config is returned.
+
+        In Autoinstall v2, unknown top-level keys result in a fatal
+        AutoinstallValidationError
+
+        Only checks for unrecognized keys, doesn't validate them.
+        Requires a version number, so this should be called after
+        validating against the schema.
+
+        """
+
+        server_config: dict[str, Any] = self._strip_controller_keys(autoinstall_config)
```

Comment:
`server_config` is a bit generic, following the restructure of `_strip_controller_keys` maybe it could be `good_keys` / `bad_keys`?